src/Dapr.Actors/Runtime/ActorTypeInformation.cs

```diff
@@ -121,7 +138,7 @@ public static ActorTypeInformation Get(Type actorType)
 
             var actorAttribute = actorType.GetCustomAttribute<ActorAttribute>();
 
-            string actorTypeName = actorAttribute?.TypeName ?? actorType.Name;
+            actorTypeName ??= actorAttribute?.TypeName ?? actorType.Name;
```

Comment:
I would prefer getting it from attribute and follow the approach which is being done for pubsub here https://github.com/dapr/dotnet-sdk/pull/681

Reply:
Sounds ok, that approach would however lock configuring actor type name to statics.

Reply:
adding @rynowak as well as he preferred the option1 which is implemented int he PR whereas I am suggesting option 2

Reply:
I think switching this to use the same technique as #681 would be fine as well. I realize now that we were talking about similar features across two different areas and I gave you different answers.

Reply:
The main difference between #681 and this is that in #681 it's working on endpoint metadata, which doesn't need to be an attribute. In this case we don't have a dependency on ASP.NET Core, thus this can't easily be built using a similar technique.

I can create a new PR, which creates a new interface `IActorMetadata`, let `ActorAttribute` implement that and then replace all references to `ActorAttribute` with `IActorMetadata`. But that will lock this implementation to only be able to use statics for overriding Actor type name. Statics should be fine for this scenario, but if one would e.g. like to use `IConfiguration` for specifying Actor type name there's a little extra (relatively ugly) work that needs to be done (e.g. create a static class and set some values prior to calling `services.AddActors(...)`).

I'll let you decide 😄

Reply:
@bjorkstromm in that case current approach seems fine, thanks for clarifying.